test/api_container_wait_test.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/alibaba/pouch/test/request"
+
+	"github.com/go-check/check"
+)
+
+// APIContainerWaitSuite is the test suite for container wait API.
+type APIContainerWaitSuite struct{}
+
+func init() {
+	check.Suite(&APIContainerWaitSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *APIContainerWaitSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+	PullImage(c, busyboxImage)
+}
+
+// TestWaitOk tests waiting a stopped container return 200.
+func (suite *APIContainerWaitSuite) TestWaitOk(c *check.C) {
+	cname := "TestWaitOk"
+
+	CreateBusyboxContainerOk(c, cname)
+	StartContainerOk(c, cname)
+	StopContainerOk(c, cname)
+
+	resp, err := request.Post("/containers/" + cname + "/wait")
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, resp, 200)
+
+	DelContainerForceOk(c, cname)
+}
+
+// TestRunningContainer tests waiting a running container to stop, then returns 200.
+func (suite *APIContainerWaitSuite) TestRunningContainer(c *check.C) {
+	cname := "TestRunningContainer"
+
+	CreateBusyboxContainerOk(c, cname)
+	StartContainerOk(c, cname)
+
+	var (
+		err  error
+		resp *http.Response
+	)
+
+	chWait := make(chan struct{})
+	go func() {
+		resp, err = request.Post("/containers/" + cname + "/wait")
+		close(chWait)
+	}()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
I think that we have problem for this case. 

`<-chWait` return which means that the channel `chWait` has been closed. In the next statement `select` , the `<-chWait` always return.

```
package main

import (
        "fmt"
        "time"
)

func main() {
        ch := make(chan struct{})

        go func() {
                time.Sleep(2 * time.Second)
                close(ch)
        }()
        fmt.Println(<-ch)
        fmt.Println(<-ch)
}
```

 So I think we still need to update the case like, WDYT?

```
chWait := make(chan struct{})
go func() {
        chWait <- struct{}{}
        resp, err = request.Post("/containers/" + cname + "/wait")
        close(chWait)
}()
<-chWait // wait for the goroutine activate
time.Sleep(100 * time.Millisecond)
stopContainerOk(c, cname)

select {
...
}
```

Reply:
since we have call `<-chWait` before, why we wait `chWait` in `select` again ?